src/services/wallets/EthWalletConnector.ts

```diff
@@ -0,0 +1,150 @@
+import { AoSigner, ContractSigner } from '@ar.io/sdk/web';
+import { MetamaskError } from '@src/utils/errors';
+import { createData, InjectedEthereumSigner } from 'arbundles';
+import { ApiConfig } from 'arweave/node/lib/api';
+import {
+  hashMessage,
+  recoverPublicKey,
+  SwitchChainError,
+  toBytes,
+  UserRejectedRequestError,
+} from 'viem';
+import { Config, Connector } from 'wagmi';
+import { connect, disconnect, getAccount, signMessage } from 'wagmi/actions';
+
+import {
+  DEFAULT_ARWEAVE_HOST,
+  DEFAULT_ARWEAVE_PORT,
+  DEFAULT_ARWEAVE_PROTOCOL,
+} from '@src/constants';
+import {
+  AoAddress,
+  NetworkPortalWalletConnector,
+  WALLET_TYPES,
+} from '../../types';
+
+export class EthWalletConnector implements NetworkPortalWalletConnector {
```

Comment:
Nit: abstract the connector logic to run off parameters to the constructor, eg connector name, then extend this base EthWalletConnector to create a MetaMaskWalletConnector.

Alternatively, if thats too much work, recommend naming this MetaMaskWalletConnector.

For example, here I have a generalized WagmiStrategy implemented, that can be configured with different wagmConfigs depending what chain or network the strategy is for. This will allow us to more easily add other signers in the future if we wish (looking at other sig types in arbundles) https://github.com/project-kardeshev/ao-wallet-kit/blob/2903b5099d10390138767e8b4f5b5ed7e9ad5e15/src/strategy/strategies/Ethereum.ts#L4